### PR TITLE
get_supported_formats: sort like the old get_preferred_format and simplify return type

### PR DESCRIPTION
**Pull request by victorvde**
**Connections**
Follow up to #2783 . Alternative to #2785 .

**Description**
Makes the first returned format the same as the old get_preferred_format. Simplifies the return type, and consequently the examples/

**Testing**
Ran some examples.

### CODE REVIEW

File: wgpu-core/src/instance.rs [hunk 1]
```diff
@@ -166,9 +166,21 @@ impl Surface {
...
+            wgt::TextureFormat::Rgba8UnormSrgb,
+            wgt::TextureFormat::Bgra8Unorm,
+            wgt::TextureFormat::Rgba8Unorm,
+            wgt::TextureFormat::Rgba16Float,
+        ];
+        caps.formats.sort_by_key(|f| {
+            preferred_formats
+                .iter()
+                .position(|pf| pf == f)
+                .unwrap_or(usize::MAX)
```
Comment (cwfitzgerald) [inline wgpu-core/src/instance.rs:182]:
```suggestion
            !f.describe().srgb
```

Per discussion

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]

File: wgpu/examples/hello-windows/main.rs [hunk 2]
```diff
@@ -33,12 +33,7 @@ impl ViewportDesc {
 
         let config = wgpu::SurfaceConfiguration {
             usage: wgpu::TextureUsages::RENDER_ATTACHMENT,
-            format: *self
-                .surface
-                .get_supported_formats(adapter)
-                .unwrap()
-                .first()
-                .unwrap(),
+            format: self.surface.get_supported_formats(adapter)[0],
```
Comment (cwfitzgerald) [inline wgpu/examples/hello-windows/main.rs:36]: Very clean!

File: wgpu/src/lib.rs [hunk 3]
```diff
@@ -3448,9 +3448,7 @@ impl Drop for SurfaceTexture {
 impl Surface {
     /// Returns a vec of supported texture formats to use for the [`Surface`] with this adapter.
     /// Note: The first format in the vector is preferred
-    ///
-    /// Returns None if the surface is incompatible with the adapter.
```
Comment (cwfitzgerald) [inline wgpu/src/lib.rs:3452]: Let's leave this note in, but refactored to represent the new situation:

```
/// Returns an empty vector if the surface is incompatible with the adapter.
```

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
One nit, otherwise lgtm

File: wgpu/src/lib.rs (hunk 3 above)
Comment (victorvde) [reply wgpu/src/lib.rs:3452]: Sure

**Pull request merged**